ext/random/random.c

```diff
@@ -354,12 +354,12 @@ PHPAPI zend_string *php_random_bin2hex_le(const void *ptr, const size_t len)
 
 /* {{{ php_random_hex2bin_le */
 /* stolen from standard/string.c */
-PHPAPI bool php_random_hex2bin_le(zend_string *hexstr, void *dest)
+PHPAPI bool php_random_hex2bin_le(const zend_string *hexstr, void *dest)
 {
 	size_t len = hexstr->len >> 1;
-	unsigned char *str = (unsigned char *) hexstr->val, c, l, d;
+	const unsigned char *str = (unsigned char *) ZSTR_VAL(hexstr);
 	unsigned char *ptr = (unsigned char *) dest;
```

Comment:
```suggestion
	unsigned char *ptr = dest;
```
Unnecessary cast from `void*`.